static/js/profile.js

```diff
@@ -0,0 +1,539 @@
+			apistatus = { 'defaults': { 'time-format': '12', 'display-units': 'mg/dL', 'language': 'en', 'custom-title': 'Nightscout'} };
+		}
+    }).done(function () {
+		$('#pe_status').hide().text('Loading profile records ...').fadeIn("slow");
+		$.ajax('/api/v1/profile.json', {
+			success: function (records) {
+				mongoprofiles = records;
+				// create new profile to be edited from last record
+				if (records[0] && records[0].dia) {
+					// Use only values(keys) defined in defaultprofile, drop the rest. Preparation for future changes.
+					profile = defaultprofile;
+					for (var key in records[0]) {
+						if (typeof profile[key] != 'undefined') 
+							profile[key] = records[0][key];
+						// copy _id of record too
+						if (key == '_id') 
+							profile[key] = records[0][key];
+						// default validfrom is current date
+						if (key == 'validfrom')
+							profile[key] = new Date();
+					}
+					$('#pe_status').hide().text('Values loaded.').fadeIn("slow");
+					mongoprofiles.unshift(profile);
+				} else {
+					profile = defaultprofile;
+					mongoprofiles.unshift(profile);
+					$('#pe_status').hide().text('Default values used.').fadeIn("slow");
+				}
+			},
+			error: function () {
+				profile = defaultprofile;
+				mongoprofiles.unshift(profile);
+				$('#pe_status').hide().text('Error. Default values used.').fadeIn("slow");
+			}
+		}).done(initeditor);
+	});
+	
+	function initeditor() {
+		// Add handler for style switching
+		$('#pe_input_simple').change(switchStyle);
+		$('#pe_input_calculator').change(switchStyle);
+
+		// display status
+		$('#pe_units').text(apistatus['defaults']['display-units']);
+		$('#pe_timeformat').text(apistatus['defaults']['time-format']+'h');
+		$('#pe_title').text(apistatus['defaults']['custom-title']);
+
+		//timepicker
+		$('#pe_validfrom').datetimepicker({
+			  lang: apistatus['defaults']['language']
+			, mask: true
+			, stepMinute: 15
+			, timeFormat: apistatus['defaults']['time-format'] == '12' ? 'hh:mm p' : 'HH:mm'
+			, dateFormat: apistatus['defaults']['time-format'] == '12' ? 'mm/dd/yy' : 'd.m.yy'
+			, minDate: mongoprofiles[1] && mongoprofiles[1].validfrom ? mongoprofiles[1].validfrom : null
+			//this has a serious bug
+			//, minDateTime: mongoprofiles[1] && mongoprofiles[1].validfrom ? mongoprofiles[1].validfrom : null
+			, maxDate: new Date()
+			, maxDateTime: new Date()
+		});
+
+		// Set values from profile to html
+		fillFrom();
+		// hide unused style of ratios
+		switchStyle();
+		
+		
+		console.log('Done initeditor()');
+	}
+	
+	// Handling html events and setting/getting values
+	function switchStyle(event) {
+		if ($('#pe_input_simple').prop('checked')) {
+			$('#pe_simple').show("slow");
+			$('#pe_calculator').hide("slow");
+		} else {
+			$('#pe_simple').hide("slow");
+			$('#pe_calculator').show("slow");
+		}
+		if (event) {
+			event.preventDefault();
+		}
+	}
+	
+	function fillFrom(event) {
+		var mmoltime = [
```

Comment:
currently is used timeformat based on env DEFAULTS. selected units are displayed on top of the page but not used for anything currently. all data are stored in raw format from html. we need to discuss format of profile JSON